endurance-page-cache.php

```diff
@@ -1143,6 +1147,7 @@ public function update( $checked_data ) {
 						}
 					}
 				}
+				set_transient( 'mojo_plugin_assets', $mu_plugin, 6 * HOUR_IN_SECONDS );
```

Comment:
@afragen This should be set immediately after the request is made using `wp_remote_get()`. Separating the setting of a transient from the getting of a transient too much can often result in bugs. For instance, this code is storing the JSON decoded HTTP response body, which is not the response returned by the `wp_remote_get()` call. This would cause the data validation checks on line 1133 to fail if we receive data from the transient. 

This line should be moved within the if condition after line 1130. I'd probably add an additional if clause to ensure we don't store a failed request. In other words, remove this line of code and take a look at the code suggestion I'm leaving for lines 1129-1131 above.

Reply:
The logic in the patch is to store the json_decode of the response body so that upon a repeat request the transient is retrieved and passed through another round of wp_remote_get. The transient is only saved if `! is_wp_error() || isset( $response['body'])`, logic reversed for explanation.

Reply:
Moving this as above is correct.

Reply:
I see. I feel like this is the classic tradeoff between clarity and cleverness. If a solution is going to reduce the understandability of the code, I'd rather make it readily apparent what is happening. I see storing a transient data value that doesn't match the value you would otherwise be fetching as a code smell. I would prefer to store the full response and let the code run up until it realizes it doesn't need to update anything.

Reply:
I think this is now moot as the transient stores the response and is doing what you think it's doing.